### ISSUE TITLE
False positive on __required__ and __optional__ on TypedDict.

### ISSUE THREAD

**Issue by rcosperec** [pylint-dev/pylint#10158]
### Bug description

```python
from typing import TypedDict, Required, NotRequired

class MyDict(TypedDict):
    required: Required[str]
    not_required: NotRequired[str]
    implicitely_required: str

def main():
    print(MyDict.__required_keys__)
    print(MyDict.__optional_keys__)

if __name__ == "__main__":
    main()
```

### Configuration

```ini

```

### Command used

```shell
pylint --disable=missing-docstring main.py
```

### Pylint output

```python
************* Module main
main.py:9:10: E1101: Class 'MyDict' has no '__required_keys__' member (no-member)
main.py:10:10: E1101: Class 'MyDict' has no '__optional_keys__' member (no-member)
```

### Expected behavior

Pylint shouldn't report it as a missing member as the code runs without any issue.

It seems the same issue was already fixed for[ __annotations__ ](https://github.com/pylint-dev/pylint/issues/7126)

### Pylint version

```shell
pylint 3.3.3
astroid 3.3.8
Python 3.11.10 (main, Sep  7 2024, 01:03:31) [Clang 15.0.0 (clang-1500.3.9.4)]
```

### OS / Environment

Darwin 24.2.0

### Additional dependencies

```python

```

**Comment by AntonWittig** [comment]
I encountered the same issue with pylint==4.0.4 as well

**Comment by TheGreatApollyon** [comment]
Fixed by https://github.com/pylint-dev/astroid/pull/3184 — the TypedDict brain now declares `__required_keys__` and `__optional_keys__`, so subclasses resolve them through the base chain.
